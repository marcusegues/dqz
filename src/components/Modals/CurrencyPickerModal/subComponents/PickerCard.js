// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { globalStyles } from '../../../../styles/globalStyles';
import type { Children } from '../../../../types/generalTypes';

const ownStyles = {
  modalContainer: {
    top: '25%',
    paddingBottom: 16,
    width: '85%',
    flexDirection: 'column',
    justifyContent: 'flex-start',
    borderRadius: 3,
    alignSelf: 'center',
    backgroundColor: '#FFF',
  },
};

type PickerCardProps = {
  children: Children,
  style: {},
};

export const PickerCard = ({ children, style }: PickerCardProps) => (
  <View
    style={[ownStyles.modalContainer, globalStyles.boxShadow, { ...style }]}
  >
    {children}
  </View>
);
