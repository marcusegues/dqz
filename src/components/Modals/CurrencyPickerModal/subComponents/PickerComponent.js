import React from 'react';
import { Picker, View } from 'react-native';

const ownStyles = {
  pickerColumn: {
    flex: 0.25,
  },
};

const PickerComponent = ({
  selectedValue,
  onValueChange,
  prompt,
  children,
  itemStyle,
  style,
}) => (
  <View style={[ownStyles.pickerColumn, { ...style }]}>
    <Picker
      selectedValue={selectedValue}
      onValueChange={onValueChange}
      mode="dropdown"
      prompt={prompt}
      itemStyle={itemStyle}
    >
      {children}
    </Picker>
  </View>
);

export default PickerComponent;
