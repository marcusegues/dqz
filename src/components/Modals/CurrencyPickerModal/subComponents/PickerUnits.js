// @flow
import React from 'react';
// $FlowFixMe
import { Text, View } from 'react-native';
import { moderateScale } from '../../../../styles/Scaling';
import { DARK_GREY } from '../../../../styles/colors';

const ownStyles = {
  pickerUnitColumn: {
    flex: 0.2,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
  pickerUnitColumnText: {
    fontFamily: 'roboto_regular',
    fontSize: moderateScale(14),
    color: DARK_GREY,
  },
};

type PickerUnitsProps = {
  unit: string,
};

export const PickerUnits = ({ unit }: PickerUnitsProps) => (
  <View style={ownStyles.pickerUnitColumn}>
    <Text style={ownStyles.pickerUnitColumnText}>{unit}</Text>
  </View>
);
