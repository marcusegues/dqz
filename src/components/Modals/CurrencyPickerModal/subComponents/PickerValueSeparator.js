// @flow
import React from 'react';
// $FlowFixMe
import { View, Text } from 'react-native';
import { moderateScale } from '../../../../styles/Scaling';

const ownStyles = {
  pickerDividerColumn: {
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
    flex: 0.1,
  },
  pickerDividerColumnText: {
    fontFamily: 'roboto_regular',
    fontSize: moderateScale(16),
    color: '#1a1a1a',
  },
};

type PickerValueSeparatorProps = {
  separator: string,
};

export const PickerValueSeparator = ({
  separator,
}: PickerValueSeparatorProps) => (
  <View style={ownStyles.pickerDividerColumn}>
    <Text style={ownStyles.pickerDividerColumnText}>{separator}</Text>
  </View>
);
