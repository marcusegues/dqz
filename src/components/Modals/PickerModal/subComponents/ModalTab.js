// @flow
import React from 'react';
// $FlowFixMe
import { Text, TouchableOpacity, View } from 'react-native';
import { moderateScale } from '../../../../styles/Scaling';
import { BLACK, GREY, MAIN_RED, WHITE } from '../../../../styles/colors';

const touchableStyles = {
  paddingVertical: 15,
  alignItems: 'center',
};

const touchableTextStyles = {
  fontFamily: 'roboto_regular',
  fontSize: moderateScale(14),
};

const ownStyles = {
  topTouchable: {
    flex: 0.5,
  },
  touchableActive: {
    ...touchableStyles,
    backgroundColor: WHITE,
    borderTopWidth: 3,
    borderTopColor: MAIN_RED,
  },
  touchableInactive: {
    ...touchableStyles,
    backgroundColor: GREY,
    borderTopWidth: 3,
    borderTopColor: GREY,
  },
  touchableActiveText: {
    ...touchableTextStyles,
    color: BLACK,
  },
  touchableInactiveText: {
    ...touchableTextStyles,
    color: WHITE,
  },
};

type ModalTabProps = {
  onPress: Function,
  standardInput: boolean,
  text: string,
};

const ModalTab = ({ standardInput, onPress, text }: ModalTabProps) => (
  <View style={ownStyles.topTouchable}>
    <TouchableOpacity
      onPress={onPress}
      style={
        standardInput ? ownStyles.touchableActive : ownStyles.touchableInactive
      }
    >
      <Text
        style={
          standardInput
            ? ownStyles.touchableActiveText
            : ownStyles.touchableInactiveText
        }
      >
        {text}
      </Text>
    </TouchableOpacity>
  </View>
);

export default ModalTab;
