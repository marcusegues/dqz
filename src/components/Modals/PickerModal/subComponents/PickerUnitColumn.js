// @flow
import React from 'react';
// $FlowFixMe
import { Text, View } from 'react-native';
import { moderateScale } from '../../../../styles/Scaling';

const ownStyles = {
  pickerUnitColumn: {
    flex: 0.2,
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
  },
  pickerUnitColumnText: {
    fontFamily: 'roboto_medium',
    fontSize: moderateScale(14),
    color: '#1A1A1A',
  },
};

type PickerUnitColumnProps = {
  unit: string,
  style?: {},
};

export const PickerUnitColumn = ({ unit, style }: PickerUnitColumnProps) => (
  <View style={[ownStyles.pickerUnitColumn, { ...style }]}>
    <Text style={ownStyles.pickerUnitColumnText}>{unit}</Text>
  </View>
);

PickerUnitColumn.defaultProps = {
  style: {},
};
