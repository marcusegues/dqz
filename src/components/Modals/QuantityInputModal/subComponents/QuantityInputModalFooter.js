// @flow
import React from 'react';
// $FlowFixMe
import { View } from 'react-native';
import { RedButton } from '../../../Buttons/RedButton';
import { pickerModalStyle } from '../../styles/PickerModal';

type QuantityInputModalFooterProps = {
  onPress: () => void,
  text: string,
};

export const QuantityInputModalFooter = ({
  onPress,
  text,
}: QuantityInputModalFooterProps) => (
  <View style={pickerModalStyle.redButtonWrapper}>
    <RedButton onPress={onPress} text={text} />
  </View>
);
