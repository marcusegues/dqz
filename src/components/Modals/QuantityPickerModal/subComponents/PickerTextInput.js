// @flow
import React from 'react';
// $FlowFixMe
import { TextInput } from 'react-native';
import { pickerModalStyle } from '../../styles/PickerModal';

export const PickerTextInput = ({ onChangeText, value }) => {
  return (
    <TextInput
      keyboardType="numeric"
      maxLength={5}
      underlineColorAndroid="transparent"
      blurOnSubmit
      style={[pickerModalStyle.textInput, { alignSelf: 'center' }]}
      onChangeText={onChangeText}
      value={value}
    />
  );
};
