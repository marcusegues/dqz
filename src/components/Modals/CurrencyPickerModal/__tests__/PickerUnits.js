import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { PickerUnits } from '../subComponents/PickerUnits';

describe('PickerUnits', () => {
  test('renders according to snapshot', () => {
    const component = renderer.create(<PickerUnits unit="kg" />).toJSON();
    expect(component).toMatchSnapshot();
  });
  test('has `kg` as a text', () => {
    const component = shallow(<PickerUnits unit="kg" />);
    const render = component.dive();
    expect(
      render
        .find(Text)
        .children()
        .text()
    ).toEqual('kg');
    expect(component).toMatchSnapshot();
  });
});
