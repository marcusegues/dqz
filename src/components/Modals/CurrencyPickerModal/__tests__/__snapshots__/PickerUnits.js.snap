// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerUnits renders according to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "flex": 0.2,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#898989",
        "fontFamily": "roboto_regular",
        "fontSize": 22,
      }
    }
  >
    kg
  </Text>
</View>
`;
